src/Sentry/SentryEvent.cs

```diff
@@ -180,22 +180,39 @@ public IReadOnlyList<string> Fingerprint
 
     internal bool HasException() => Exception is not null || SentryExceptions?.Any() == true;
 
-    internal bool HasTerminalException()
+    internal bool HasUnhandledException()
     {
-        // The exception is considered terminal if it is marked unhandled,
-        // UNLESS it comes from the UnobservedTaskExceptionIntegration
+        if (Exception?.Data[Mechanism.HandledKey] is false)
+        {
+            return true;
+        }
+
+        return SentryExceptions?.Any(e => e.Mechanism is { Handled: false }) ?? false;
+    }
+
+    internal bool HasUnhandledNonTerminalException()
```

Comment:
question: order

Do I get that right that, for correct consumption, we need to call these methods in this order
1. `HasUnhandledNonTerminalException`
2. `HasUnhandledException`
3. `HasException`

Then there is also `HasTerminalException`.

It's a bit confusing to me ... I wonder if this could be designed a bit more clearly.
On the other hand ... this may just be me ... and the methods are all `internal`.

Reply:
I see. I wrapped the thing with an `enum ExceptionType` and make the rest `private`. That makes it easier to reason about.